### PR TITLE
Update chatbot for slack integration

### PR DESCRIPTION
**Pull request by JReko**
I believe all that PR does is properly listed in the commit list. A few simple changes to already existing stuff, almost all of them are to access some data in different classes. Added a cache interface for possibility to use memcached or something else in the future. For the moment, added a "poorman's cache" allowing the daemon to store the data in plain memory.

### CODE REVIEW

**Review by DaazKu — APPROVED** [review]

File: src/Bot/Command/AbstractCommand.php [hunk 1]
```diff
@@ -8,6 +8,7 @@
 namespace Kaecyra\ChatBot\Bot\Command;
 use Kaecyra\AppCommon\Event\EventAwareInterface;
 use Kaecyra\AppCommon\Event\EventAwareTrait;
+use Kaecyra\ChatBot\Client\Slack\Strategy\AbstractStrategy;
```
Comment (kaecyra) [inline src/Bot/Command/AbstractCommand.php]: I don't like this. Makes Bot depend on Slack.

**Review by kaecyra — CHANGES_REQUESTED** [review]

File: src/Bot/Command/AbstractCommand.php (hunk 1 above)
Comment (JReko) [reply src/Bot/Command/AbstractCommand.php]: Hmm, I agree, the thing is, it was already implicitly depending on Bot in a way.

https://github.com/kaecyra/chatbot/blob/7d11cd5e60c38fe21b2024dcbb7d5dd166cac2be/src/Client/Slack/SlackRtmClient.php#L644-L675

That `->strategy` property is used dynamically. The question I'm asking myself right now is should strategies really be under the slack namespace? I mean they aren't really client related?

Comment (JReko) [reply src/Bot/Command/AbstractCommand.php]: Will move the abstract strategy under Bot.

**Review by kaecyra — APPROVED** [review]
Looks good Frank

**Pull request merged**